test/unit/test_handler_mocking_openidc.lua

```diff
@@ -405,6 +406,45 @@ function TestHandler:test_authenticate_with_session_cookie_samesite_set_to_none(
   lu.assertItemsEquals(v, opts.session)
 end
 
+function TestHandler:test_authenticate_ok_with_auth_request()
+  -- arrange
+  local actual_unauth_action
+  kong.request.get_path = function()
+    return idpAuthPath
+  end
+
+  -- mock authenticate to be able to check unauth_action
+  self.module_resty.openidc.authenticate = function(opts, target_url, unauth_action)
+    actual_unauth_action = unauth_action
+    return {}, false, "/", session
+  end
+  -- act
+  self.handler:access({ idp_authentication_path = idpAuthPath })
+
+  -- assert
+  lu.assertTrue(self:log_contains("login request detected"))
+  lu.assertEquals(actual_unauth_action, nil)
+end
+
+function TestHandler:test_authenticate_ok_with_api_request()
```

Comment:
@iankoplowitzdev  change `_api_request` to `_xmlhttprequest`.

Reply:
This method will check any api request, not just Ajax calls.